### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by Tracy-miranja**
- [ ] Interface- [ ] Interface
- [ ] Create a class Nameable.
- [ ] Implement a method called correct_name that will raise a NotImplementedError.
- [ ] Turn your Person class to Nameable
- [ ] Make sure that your Person class inherits from Nameable
- [ ] Make sure that this class has a method correct_name implemented. It should simply return the name attribute.
- [ ] Prepare base Decorator
- [ ] Make sure that it inherits from Nameable.
- [ ] In the constructor assign a nameable object from params to an instance variable.
- [ ] Implement the correct_name method that returns the result of the correct_name method of the @nameable.
- [ ] Prepare CapitalizeDecorator and TrimmerDecorator
- [ ] For the CapitalizeDecorator:
- [ ] Create a class that inherits from the base Decorator class.
- [ ] Implement a method correct_name that capitalizes the output of @nameable.correct_name.
- [ ] For the TrimmerDecorator:
- [ ] Create a class that inherits from the base Decorator class.
- [ ] Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.
- [ ] Create a class Nameable.
- [ ] Implement a method called correct_name that will raise a NotImplementedError.
- [ ] Turn your Person class to Nameable
- [ ] Make sure that your Person class inherits from Nameable
- [ ] Make sure that this class has a method correct_name implemented. It should simply return the name attribute.
- [ ] Prepare base Decorator
- [ ] Make sure that it inherits from Nameable.
- [ ] In the constructor assign a nameable object from params to an instance variable.
- [ ] Implement the correct_name method that returns the result of the correct_name method of the @nameable.
- [ ] Prepare CapitalizeDecorator and TrimmerDecorator
- [ ] For the CapitalizeDecorator:
- [ ] Create a class that inherits from the base Decorator class.
- [ ] Implement a method correct_name that capitalizes the output of @nameable.correct_name.
- [ ] For the TrimmerDecorator:
- [ ] Create a class that inherits from the base Decorator class.
- [ ] Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

- [ ] See your decorators in action

\Try the following code and check if you managed to decorate your person:

person = Person.new(22, 'maximilianus')
  person.correct_name
  capitalizedPerson = CapitalizeDecorator.new(person)
  capitalizedPerson.correct_name
  capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
  capitalizedTrimmedPerson.correct_name

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require './namable_class'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
```
Comment (OLIPLICHE) [inline base_decorator.rb]: - Hi, Reading from the project requirements: "In the constructor assign a nameable object from params to an instance variable", I would advise you to define` nameable` before moving to initialization. Therefore add this code: `attr_accessor :nameable`

File: base_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
```
Comment (OLIPLICHE) [inline base_decorator.rb]: - [ ] In reading your project requirements, it says: "**For the CapitalizeDecorator**:
Create a class that inherits from the base Decorator class." 
- Therefore, I would advise you to create CapitalizeDecorator class separately and move this code inside

File: base_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    name = @nameable.correct_name.capitalize
+    name.length > 10 ? name[0..9] : name
+  end
+end
```
Comment (OLIPLICHE) [inline base_decorator.rb]: - [ ] In reading your project requirements, it says: "For the TrimmerDecorator:
Create a class that inherits from the base Decorator class."
Therefore, I would advise you to create TrimmerDecorator class separately and move this code inside.

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Tracy-miranja, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Create a class Nameable 👍 
- Profesional pool request 👍
- No linter errors 👍 

## Required Changes ♻️
- I would advise you to check if you have managed to decorate your person before asking for a re-review, as it is stated in the requirements
_**Check the comments under the review**._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @Tracy-miranja 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - All changes requested have been implemented 🥇  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Tracy-miranja** [comment]
Thanks for the review

**Pull request merged**